packages/astro/src/vite-plugin-astro/index.ts

```diff
@@ -123,10 +123,10 @@ ${err.url}`;
         throw err;
       }
     },
-    async handleHotUpdate(context) {
-      if (devServer) {
-        return devServer.handleHotUpdate(context);
-      }
-    },
+    // async handleHotUpdate(context) {
```

Comment:
This PR disables our HMR for Vite to do more. Couldn’t find a quick fix

Reply:
Thanks, I had run into the same issues over the past month. It's a tough call but I think it's better to ship this when it's ready, vs. as a part of v0.21 with known bugs. Vite's default HMR is still pretty great on its own, all we lose here (afaik) is the smart updating HTML on an Astro component change.